### PR TITLE
Conventions and Rubocop post

### PR DESCRIPTION
**Pull request by rdormer**

### CODE REVIEW

File: _posts/2019-06-07-conventions-and-rubocop.markdown [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+date: 2019-06-07 09:00:00
+categories: ["ruby", "best-practices"]
+author: "rdormer"
+---
+
+Everyone has had the experience of working on a gnarly, difficult to
+understand code-base.  The sort of code base that makes you hate your
+job.  Often it comes down to poor design, but [code
+conventions](https://en.wikipedia.org/wiki/Coding_conventions) also
+play a large part in wether you wake up dreading your job in the
```
Comment (lubc) [inline _posts/2019-06-07-conventions-and-rubocop.markdown]:
```suggestion
play a large part in whether you wake up dreading your job in the
```

File: _posts/2019-06-07-conventions-and-rubocop.markdown [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+idea of being restricted, it's unquestionably easier to write code
+when you don't have to spend a lot of time thinking about how to name
+variables or classes (for instance) because there's really only one
+way you're allowed to do it.
+
+Of course, developers love to disagree, and any conventions like this
+will inevitably be a potential source of griping.  From experience,
+it'll be a fraction of the complaining you'll hear if you *don't* have
+standard conventions, though.  Even the most opinionated developers
+will be professional enough to get used to it quickly enough.  Wether
```
Comment (lubc) [inline _posts/2019-06-07-conventions-and-rubocop.markdown]:
```suggestion
will be professional enough to get used to it quickly enough.  Whether
```

File: _posts/2019-06-07-conventions-and-rubocop.markdown [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+integrate Rubocop into the picture, you'll get dozens if not hundreds
+of style violations.  No worries, Rubocop has you covered.  The handy
+[autocorrect](https://rubocop.readthedocs.io/en/latest/auto_correct/)
+mode will fix violations automatically.  If that's a little aggressive
+for your tastes, then you can just generate a
+[rubocop_todo.yml](https://rubocop.readthedocs.io/en/latest/configuration/#automatically-generated-configuration)
+file that will disable checking for all *existing* violations while
+still flagging any new violations going forward.  You can then circle
+back and fix up your existing code, treating it as a part of your
+existing technical debt.
```
Comment (etagwerker) [inline _posts/2019-06-07-conventions-and-rubocop.markdown]: @rdormer Could you add a section about `standardrb`? https://github.com/testdouble/standard

File: _posts/2019-06-07-conventions-and-rubocop.markdown [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+variables or classes (for instance) because there's really only one
+way you're allowed to do it.
+
+Of course, developers love to disagree, and any conventions like this
+will inevitably be a potential source of griping.  From experience,
+it'll be a fraction of the complaining you'll hear if you *don't* have
+standard conventions, though.  Even the most opinionated developers
+will be professional enough to get used to it quickly enough.  Wether
+they admit it or not, most experienced developers would rather have a
+convention they're not crazy about than no conventions at all.
```
Comment (etagwerker) [inline _posts/2019-06-07-conventions-and-rubocop.markdown]: @rdormer Maybe you could add a paragraph about discussions in code reviews? I believe having an automated tool for this we will save time in code style discussions.

**Pull request merged**